CMakePresets.json

```diff
@@ -0,0 +1,86 @@
+{
+    "version"   : 3,
+    "cmakeMinimumRequired" : {
+        "major" : 3,
+        "minor" : 22,
+        "patch" : 0
+    },
+    "configurePresets" : [
+        {
+            "name"          : "default",
+            "binaryDir"     : "${sourceDir}/build-with-${presetName}",
+            "cacheVariables" : {
+                "CMAKE_EXPORT_COMPILE_COMMANDS"     : {"type" : "BOOL"  , "value" : "ON"},
+                "CMAKE_BUILD_TYPE"                  : {"type" : "STRING", "value" : "Release"},
+                "CMAKE_CXX_STANDARD"                : {"type" : "STRING", "value" : "17"},
+                "KokkosTools_REUSE_KOKKOS_COMPILER" : {"type" : "BOOL"  , "value" : "ON"},
```

Comment:
Sorry, it is unclear what you're pointing to. Are you referring to `KokkosTools_REUSE_KOKKOS_COMPILER`?

Reply:
Yes

Reply:
Ok :)

The idea of the PR is to add a CI for joint `Kokkos` / `Kokkos Tools` builds. `KokkosTools_REUSE_KOKKOS_COMPILER` is a `Kokkos Tools` (advanced) option that forces to reuse the `Kokkos` compiler (e.g. `nvcc_wrapper` for Cuda I guess).

I thought it would be nice to ensure that during a joint build, both projects are on the same page for the compiler.

What's your opinion on that?